### PR TITLE
[Issue-578] Improving the compaction performance.

### PR DESCRIPTION
**Pull request by ravikiran23**
1. made the limit of 120000 in memory size in case of compaction.
2. reducing the number of tasks launched in case of compaction.

### CODE REVIEW

**Comment by ravikiran23** [comment]
retest this please

File: core/src/main/java/org/carbondata/core/carbon/datastore/block/TableTaskInfo.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        if (null == nodeOccurence) {
+          nodeAndOccurenceMapping.put(node, 1);
+        } else {
+          nodeOccurence++;
+        }
+      }
+    }
+    Integer previousValueOccurence = null;
+
+    for (Map.Entry<String, Integer> entry : nodeAndOccurenceMapping.entrySet()) {
```
Comment (gvramana) [inline core/src/main/java/org/carbondata/core/carbon/datastore/block/TableTaskInfo.java]: Add comment

File: integration/spark/src/main/scala/org/apache/spark/sql/CarbonContext.scala [hunk 2]
```diff
@@ -150,4 +151,18 @@ object CarbonContext {
   }
 
   def datasourceName: String = "org.apache.carbondata.format"
+
```
Comment (gvramana) [inline integration/spark/src/main/scala/org/apache/spark/sql/CarbonContext.scala]: Write comment

File: integration/spark/src/main/scala/org/carbondata/spark/rdd/CarbonMergerRDD.scala [hunk 3]
```diff
@@ -194,40 +205,75 @@ class CarbonMergerRDD[K, V](
           inputSplit.getLocations, inputSplit.getLength
         )
       )
+
```
Comment (gvramana) [inline integration/spark/src/main/scala/org/carbondata/spark/rdd/CarbonMergerRDD.scala]: Add comments

File: integration/spark/src/main/scala/org/carbondata/spark/rdd/CarbonMergerRDD.scala [hunk 4]
```diff
@@ -194,40 +205,75 @@ class CarbonMergerRDD[K, V](
...
-
-    // combining the segments output i.e combining list of maps.
-    val nodeBlockMapping: util.Map[String, util.List[TableBlockInfo]] =
-      CarbonDataMergerUtil.combineNodeBlockMaps(mapsOfNodeBlockMapping)
-
-
+      nodeMapping =
+        CarbonLoaderUtil.nodeBlockMapping(taskInfoList, -1)
+
+    val confExecutors = confExecutorsTemp.toInt
```
Comment (gvramana) [inline integration/spark/src/main/scala/org/carbondata/spark/rdd/CarbonMergerRDD.scala]: move this function to DistributionUtil
ensureExecutorsAndGetNodeList

File: integration/spark/src/main/scala/org/carbondata/spark/rdd/Compactor.scala [hunk 5]
```diff
@@ -83,12 +83,14 @@ object Compactor {
     carbonLoadModel.setLoadMetadataDetails(segmentStatusManager
       .readLoadMetadata(carbonTable.getMetaDataFilepath()).toList.asJava
     )
+    val execInstance = sc.sparkContext.getConf.get("spark.executor.instances", "3")
```
Comment (gvramana) [inline integration/spark/src/main/scala/org/carbondata/spark/rdd/Compactor.scala]: make default value to 1

File: integration/spark/src/main/scala/org/carbondata/spark/rdd/CarbonMergerRDD.scala [hunk 6]
```diff
@@ -194,40 +205,75 @@ class CarbonMergerRDD[K, V](
...
-    nodeBlockMapping.asScala.foreach { entry =>
-      val blocksPerNode = entry._2
-      if (blocksPerNode.size() != 0) {
-        result.add(new CarbonSparkPartition(id, i, Seq(entry._1).toArray, blocksPerNode))
-        i += 1
-      }
+    nodeMapping.asScala.foreach { entry =>
+
+      val list = new util.ArrayList[TableBlockInfo]
+      entry._2.asScala.foreach(taskInfo => {
```
Comment (gvramana) [inline integration/spark/src/main/scala/org/carbondata/spark/rdd/CarbonMergerRDD.scala]: print task distribution info

File: core/src/main/java/org/carbondata/core/carbon/datastore/block/TableTaskInfo.java (hunk 1 above)
Comment (ravikiran23) [inline core/src/main/java/org/carbondata/core/carbon/datastore/block/TableTaskInfo.java]: fixed

File: integration/spark/src/main/scala/org/apache/spark/sql/CarbonContext.scala (hunk 2 above)
Comment (ravikiran23) [inline integration/spark/src/main/scala/org/apache/spark/sql/CarbonContext.scala]: fixed

File: integration/spark/src/main/scala/org/carbondata/spark/rdd/Compactor.scala (hunk 5 above)
Comment (ravikiran23) [inline integration/spark/src/main/scala/org/carbondata/spark/rdd/Compactor.scala]: fixed

File: integration/spark/src/main/scala/org/carbondata/spark/rdd/CarbonMergerRDD.scala (hunk 4 above)
Comment (ravikiran23) [inline integration/spark/src/main/scala/org/carbondata/spark/rdd/CarbonMergerRDD.scala]: done

File: integration/spark/src/main/scala/org/carbondata/spark/rdd/CarbonMergerRDD.scala (hunk 3 above)
Comment (ravikiran23) [inline integration/spark/src/main/scala/org/carbondata/spark/rdd/CarbonMergerRDD.scala]: done

**Pull request merged**